net/dns/manager.go

```diff
@@ -137,6 +139,47 @@ func (m *Manager) Set(cfg Config) error {
 	return nil
 }
 
+// compileHostEntries creates a list of single-label resolutions possible
+// from the configured hosts and search domains.
+// The entries are compiled in the order of the search domains, then the hosts.
+// The returned list is sorted by the first hostname in each entry.
+func compileHostEntries(cfg Config) (hosts []*HostEntry) {
+	didLabel := make(map[string]bool, len(cfg.Hosts))
+	for _, sd := range cfg.SearchDomains {
+		for h, ips := range cfg.Hosts {
+			if !sd.Contains(h) || h.NumLabels() != (sd.NumLabels()+1) {
+				continue
+			}
+			ipHosts := []string{string(h.WithTrailingDot()), ""}[:1]
```

Comment:
Mysterious. I can reverse engineer that you're saving an allocation further down, but it's a really unusual construct. Comment, at least?

Reply:
Also arguably unnecessary. We don't reconfigure DNS that often, a little garbage should be fine? I don't mind keeping it, it's just a bit out of place since this isn't super hot code.